R/get_term_labels.R

```diff
@@ -0,0 +1,26 @@
+get_term_labels <- function(x, idx = NULL) {
+  if (is.data.frame(x)) {
+    if ("term" %in% names(x) && length(unique(x$term)) == nrow(x)) {
+      return(unique(x$term))
+    } else if (any(grepl("^contrast", names(x)))) {
+      tmp <- grep("^term$|^contrast", names(x))
+      out <- x[, tmp, drop = FALSE]
+      if (length(unique(out[["term"]])) == 1) {
+        out[["term"]] <- NULL
+      }
+      out <- apply(out, 1, paste, collapse = " ")
```

Comment:
Just dropping by to say that this is faster and more memory efficient:
```r
out <- do.call(paste, c(out, sep = " "))
```

Reply:
Old habits die hard. I now remember you teaching me this trick already. :)

Reply:
Yeah, I now have an alarm in my head for this one 😄